### PR TITLE
portlist: wait for lsof cmd to exit

### PR DESCRIPTION
**Pull request by maisem**
We were leaking processes otherwise.

Co-authored-by: Mihai Parparita <mihai@tailscale.com>
Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

File: portlist/portlist_macos.go [hunk 1]
```diff
@@ -159,6 +159,9 @@ func (im *macOSImpl) addProcesses() error {
 		}
 		return nil
 	}
+	defer lsofCmd.Process.Wait()
```
Comment (mihaip) [inline portlist/portlist_macos.go:162]: Can we look at the `ProcessState.ExitCode()` of this and use that to populate `lsofFailed` instead? That appears to be the sandbox failure mode (vs. the whole process not starting).

**Review by mihaip — COMMENTED** [review]
Typo with "exist" in the PR/commit title.

File: portlist/portlist_macos.go (hunk 1 above)
Comment (maisem) [reply portlist/portlist_macos.go]: Done. PTAL

**Review by mihaip — APPROVED** [review]

**Review by bradfitz — APPROVED** [review]

File: portlist/portlist_macos.go [hunk 2]
```diff
@@ -159,6 +159,14 @@ func (im *macOSImpl) addProcesses() error {
 		}
 		return nil
 	}
+	defer func() {
+		ps, err := lsofCmd.Process.Wait()
+		if err != nil || ps.ExitCode() != 0 {
```
[the remark this answers is not in the document]
Comment (maisem) [reply portlist/portlist_macos.go:164]: Done

**Pull request merged**